### PR TITLE
Update facet filtering to only allow one facet filter per request

### PR DESCRIPTION
**Pull request by chejennifer**

### CODE REVIEW

File: proto/v2/observation.proto [hunk 1]
```diff
@@ -62,11 +62,9 @@ message FacetObservation {
...
 message FacetFilter {
-  // List of variables that this facet filter applies to
-  repeated string variables = 1;
-  // When specified, only observations with provenance in this domain is
+  // When specified, only observations with provenance in these domains are
   // returned.
-  string domain = 2;
+  repeated string domains = 2;
   // When specified, only observations with this facet id are returned
   string facet_id = 3;
```
Comment (shifucun) [inline proto/v2/observation.proto:69]: i thought you want to keep repeated facet_id?

File: internal/server/v2/observation/golden/contained_in_2015_test.go [hunk 2]
```diff
@@ -116,7 +97,7 @@ func TestFetchContainIn(t *testing.T) {
 			if latencyTest {
 				continue
 			}
-			if test.GenerateGolden {
+			if true {
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply internal/server/v2/observation/golden/contained_in_2015_test.go]: done

File: internal/server/v2/observation/golden/contained_in_2015_test.go [hunk 3]
```diff
@@ -40,64 +40,45 @@ func TestFetchContainIn(t *testing.T) {
...
+			filter           *pbv2.FacetFilter
 			goldenFile       string
 		}{
 			{
 				[]string{
 					"test_var_1",
 				},
 				"country/USA<-containedInPlace+{typeOf:State}",
-				[]*pbv2.FacetFilter{},
+				&pbv2.FacetFilter{},
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply internal/server/v2/observation/golden/contained_in_2015_test.go]: done

File: internal/server/v2/observation/golden/direct_test.go [hunk 4]
```diff
@@ -186,7 +173,7 @@ func TestFetchDirect(t *testing.T) {
 			if latencyTest {
 				continue
 			}
-			if test.GenerateGolden {
+			if true {
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply internal/server/v2/observation/golden/direct_test.go]: done

File: proto/v2/observation.proto (hunk 1 above)
Comment (chejennifer) [reply proto/v2/observation.proto]: chatted offline and changed to repeated

**Review by shifucun — APPROVED** [review]
nice!

**Pull request merged**